### PR TITLE
#7 setup modules packages, skeleton for hlsp

### PR DESCRIPTION
**Pull request by rmelian**
in this first PR I'm setting up packages and started working on a skeleton for the hlsp module

NOTE: this is still WIP

image

### CODE REVIEW

File: hlsp.go [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+	return
+}
+
+func convertToJSON(doc string) (string, error) {
+	if isJSON(doc) {
+		// convert here JSON to YML
+		return "", nil
+	}
+
+	if isYML(doc) {
```
Comment (MikeRalphson) [inline hlsp.go]: Please can we use YAML/yaml everywhere not YML/yml, which doesn't really mean anything unless you're using DOS or Windows 3.

File: hlsp.go [hunk 2]
```diff
@@ -1,6 +1,30 @@
...
 // Parse receives either a YAML or JSON AsyncAPI document. It parses the document and checks if it's valid AsyncAPI. Skips specification extensions and schemas validation. If validation fails, the Parser/Validator should trigger an error. Produces a beautified version of the document in JSON Schema Draft 07
-func Parse() string {
-	return "Hello World"
+func Parse(document string) (string, err error) {
+
+	return
+}
+
+func convertToJSON(doc string) (string, error) {
+	if isJSON(doc) {
```
Comment (MikeRalphson) [inline hlsp.go]: Is it important to know if the input doc is in JSON format? JSON is a subset of YAML.

File: hlsp.go [hunk 3]
```diff
@@ -1,6 +1,30 @@
...
+		// convert here JSON to YML
+		return "", nil
+	}
+
+	if isYML(doc) {
+		return doc, nil
+	}
+
+	return "", fmt.Errorf("Not supported Document format. Supported formats: JSON or YAML")
```
Comment (MikeRalphson) [inline hlsp.go]: "Not a supported" or "Unsupported".

**Comment by fmvilas** [comment]
Oops, I just saw your comments @MikeRalphson. We don't need to know if it's YAML, but we do need to know if it's JSON. I'm changing it.

**Comment by fmvilas** [comment]
Ok! I made sure this compiles! 😄 It doesn't need to have any "value", but just setting up the project basics.

#### Tip
```
go build ./...

go test ./...
```

#### Some good documentation
* https://github.com/golang-standards/project-layout
* https://github.com/golang/go/wiki/Modules

File: sep.go [hunk 4]
```diff
@@ -1,6 +0,0 @@
-package hlsp
```
Comment (fmvilas) [inline sep.go]: This file is not needed because we will not write specification extensions. They will be (optionally) passed to the parser.Parse method.

**Comment by fmvilas** [comment]
I'm merging this PR to unblock the other issues. Please, don't take this merge as a "conversation is done" 🙏

File: sep.go (hunk 4 above)
Comment (rmelian) [reply sep.go]: I had no idea about go.mod, very good this

**Pull request merged**